relay-server/src/processing/trace_metrics/process.rs

```diff
@@ -100,24 +100,26 @@ fn scrub_trace_metric(metric: &mut Annotated<TraceMetric>, ctx: Context<'_>) ->
 
 /// Normalizes an individual trace metric entry.
 fn normalize_trace_metric(metric: &mut Annotated<TraceMetric>, meta: &RequestMeta) -> Result<()> {
-    let Some(metric_value) = metric.value_mut() else {
-        return Err(Error::Invalid(DiscardReason::InvalidTraceMetric));
+    if let Some(metric_value) = metric.value_mut() {
+        eap::normalize_received(&mut metric_value.attributes, meta.received_at());
+        eap::normalize_user_agent(
+            &mut metric_value.attributes,
+            meta.user_agent(),
+            meta.client_hints(),
+        );
+        eap::normalize_attribute_types(&mut metric_value.attributes);
     };
 
-    eap::normalize_received(&mut metric_value.attributes, meta.received_at());
-    eap::normalize_user_agent(
-        &mut metric_value.attributes,
-        meta.user_agent(),
-        meta.client_hints(),
-    );
-    eap::normalize_attribute_types(&mut metric_value.attributes);
+    process_value(
+        metric,
+        &mut SchemaProcessor::new().with_required(RequiredMode::DeleteParent),
+        ProcessingState::root(),
+    )?;
 
-    if metric_value.ty.value().is_none() {
-        return Err(Error::Invalid(DiscardReason::InvalidTraceMetric));
+    if let Annotated(None, meta) = metric {
+        relay_log::debug!("empty metric: {meta:?}");
+        return Err(Error::Invalid(DiscardReason::NoData));
```

Comment:
Not sure how easy this would be, but could we pull the reason from `meta`? E.g. if it's "missing attribute" we log "missing attribute" as discard reason? `"no_data" seems a bit misleading to the consumer of the outcome.